compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -1279,19 +1279,24 @@ fn generic_simd_intrinsic<'ll, 'tcx>(
     }
 
     if name == sym::simd_shuffle {
-        // Make sure this is actually an array, since typeck only checks the length-suffixed
+        // Make sure this is actually an array or SIMD vector, since typeck only checks the length-suffixed
         // version of this intrinsic.
-        let n: u64 = match args[2].layout.ty.kind() {
+        let idx_ty = args[2].layout.ty;
```

Comment:
"index type"?
isn't it more a "type at this index", so "this type's index"? Doesn't matter much, just curious.

Reply:
It's the type of the argument named `idx` in the [signature](https://doc.rust-lang.org/nightly/std/intrinsics/simd/fn.simd_shuffle.html).